roles/kubernetes/preinstall/tasks/0020-set_facts.yml

```diff
@@ -230,10 +230,14 @@
   set_fact:
     nameserverentries: |-
       {{ (nameservers | d([]) + cloud_resolver | d([]) + configured_nameservers | d([])) | unique | join(',') }}
-    supersede_nameserver:
-      supersede domain-name-servers {{ (nameservers | d([]) + cloud_resolver | d([])) | unique | join(', ') }};
+    dhclient_supersede_nameserver_entries_list: |-
+      {{ (nameservers | d([]) + cloud_resolver | d([])) | unique }}
   when: dns_early and not dns_late
 
+- name: Generate supersede_nameserver from dhclient_supersede_nameserver_entries_list
+  set_fact:
+    supersede_nameserver: "{{ ('supersede domain-name-servers ' + (dhclient_supersede_nameserver_entries_list | join(', ')) + ';') if dhclient_supersede_nameserver_entries_list | length > 0  else '' }}"
```

Comment:
I feel like this could be a multiline template for readability with something like that:
```
{%- if dhclient_supersede_nameserver_entries_list | length > 0 -%}
supersede domain-name-servers  {{ dhclient_supersede_nameserver_entries_list | join(', ') }};
{%- endif -%}
```

WDYT?

Reply:
Agreed!

In f74eef64c0c82b7adc28adfdb677f8c5c8003530, I've switch to a multiline template as proposed.